geode-redis/src/main/java/org/apache/geode/redis/internal/data/RedisStringCommandsFunctionExecutor.java

```diff
@@ -63,30 +63,30 @@ public boolean set(ByteArrayWrapper key, ByteArrayWrapper value, SetOptions opti
 
   @Override
   public long incr(ByteArrayWrapper key) {
-    return stripedExecute(key, () -> getRedisString(key).incr(getRegion(), key));
+    return stripedExecute(key, () -> getRedisString(key, false).incr(getRegion(), key));
   }
 
   @Override
   public long decr(ByteArrayWrapper key) {
-    return stripedExecute(key, () -> getRedisString(key).decr(getRegion(), key));
+    return stripedExecute(key, () -> getRedisString(key, false).decr(getRegion(), key));
   }
 
   @Override
   public ByteArrayWrapper getset(ByteArrayWrapper key, ByteArrayWrapper value) {
     return stripedExecute(key,
-        () -> getRedisString(key).getset(getRegion(), key, value));
+        () -> getRedisString(key, false).getset(getRegion(), key, value));
```

Comment:
`@bob` This is wrong - it should be `true` here.

Reply:
> The story says that we should only update stats for those commands that retrieve data.

Reply:
`@bob` Well, this command does both. It `get`s a value and then `set`s a new value so we do need to update the stat. Did you even check how native redis handles this before assuming you knew what to do here?